packages/cli/src/PublicApi/v1/handlers/workflows/workflows.handler.ts

```diff
@@ -230,62 +243,70 @@ export = {
 		async (req: WorkflowRequest.Activate, res: express.Response): Promise<express.Response> => {
 			const { id } = req.params;
 
-			const sharedWorkflow = await getSharedWorkflow(req.user, id);
+			const workflow = await Container.get(SharedWorkflowRepository).findWorkflowForUser(
+				id,
+				req.user,
+				['workflow:update'],
+			);
```

Comment:
> Would it make sense to augment the existing authorize(...) call to also check for the scopes, instead of passing it as part of the function call? 

I could take a look, but I think it would just create more conflicts with what @valya is doing. 

> This would bring the public API closer to the internal.

The internal API is using the scope decorator, but additionally also uses `findWorkflowForUser` in a lot of places, because in a service function I just can't be sure that I was called by a controller that already checked the permissions for the workflow/credential I'm going to access.

Reply:
Ah gotcha, maybe we can address this when changing the public API.

I'm afraid all those changes might leave a bit of tech debt due to inconsistencies and I'd like to avoid that if possible.

Reply:
> I'm afraid all those changes might leave a bit of tech debt due to inconsistencies and I'd like to avoid that if possible.

I can make the additional changes to the `authorize` function. 

I would refactor this to be able to reuse it for this:
https://github.com/n8n-io/n8n/blob/3875d358e72e46328c65ac0f09d7eeca67d60876/packages/cli/src/decorators/registerController.ts#L54-L148

WDYT?

Reply:
We agreed on not touching this to avoid conflicts with Val's branch.